WNPRC_Purchasing/src/client/components/LineItemsPanel.tsx

```diff
@@ -50,67 +53,94 @@ export const LineItemsPanel: FC<Props> = memo(props => {
+                                <Col xs={!hasRequestId ? 2 : 1}>Controlled substance</Col>
+                                { hasRequestId && (isAdmin || canUpdate) &&
+                                    <Col xs={1}>Quantity received</Col>
+                                }
+                            </Row>
+                        }
+                    </div>
+                    <div>
+                        {lineItems.map((lineItem, idx) => {
+                            return (
+                                <LineItemRow
+                                    rowIndex={idx}
+                                    key={'line-item-' + idx}
+                                    model={lineItem}
+                                    onInputChange={lineItemRowChange}
+                                    onDelete={onDeleteRow}
+                                    isAdmin={isAdmin}
+                                    canUpdate={canUpdate}
+                                    canInsert={canInsert}
+                                    hasRequestId={hasRequestId}
+                                />
+                            );
+                        })}
+                    </div>
+                    <div>
+                        {
+                            (isAdmin || !canUpdate) && (
+                                <>
+                                    <div>
+                                        <Row>
+                                            <Col xs={4}></Col>
+                                            <Col xs={1}></Col>
+                                            <Col xs={1}></Col>
+                                            <Col className="calc-total" xs={1}>
+                                                Total ($):
+                                            </Col>
+                                            <Col className="calc-total-val" xs={1}>
+                                                {formatCurrency(
+                                                    lineItems.reduce(function (accumulatedVal, lineItem) {
+                                                        return lineItem.unitCost && lineItem.quantity
+                                                            ? accumulatedVal + lineItem.unitCost * lineItem.quantity
+                                                            : accumulatedVal;
+                                                    }, 0)
+                                                )}
+                                            </Col>
+                                            <Col xs={1}></Col>
+                                            <Col xs={1}></Col>
+                                        </Row>
+                                    </div>
+                                    <div className="add-item">
+                                        <span id="add-line-item-row" title="Add item" className="add-item-icon" onClick={onClickAddRow}>
```

Comment:
onClickAddRow above needs to be a useCallback.

Reply:
done